File: app/src/main/java/com/example/massa/luxvilla/adaptadores/adaptadorrvtodas.java
package com.example.massa.luxvilla.adaptadores;


import android.content.Context;
import android.content.SharedPreferences;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.android.volley.VolleyError;
import com.android.volley.toolbox.ImageLoader;
import com.daimajia.androidanimations.library.Techniques;
import com.daimajia.androidanimations.library.YoYo;
import com.example.massa.luxvilla.R;
import com.example.massa.luxvilla.network.VolleySingleton;
import com.example.massa.luxvilla.utils.firebaseutils;
import com.example.massa.luxvilla.utils.todascasas;
import com.like.IconType;
import com.like.LikeButton;
import com.like.OnLikeListener;

import java.util.ArrayList;

/**
 * Created by massa on 22/01/2016.
 */
public class adaptadorrvtodas extends RecyclerView.Adapter<adaptadorrvtodas.vhtodas> {

    private ArrayList<todascasas> casas=new ArrayList<>();
    private LayoutInflater layoutInflater;
    private ImageLoader imageLoader;
    public static Context ctx;

    public  adaptadorrvtodas(Context context){
        layoutInflater= LayoutInflater.from(context);
        VolleySingleton volleySingleton=VolleySingleton.getInstancia(context);
        imageLoader=volleySingleton.getImageLoader();
        ctx=context;

    }

    public void setCasas(ArrayList<todascasas> cs){
        this.casas=cs;
        notifyDataSetChanged();
    }

    public void removeAt(int position) {
        casas.remove(position);
        notifyItemRemoved(position);
        notifyItemRangeChanged(position, casas.size());
    }

    @Override
    public vhtodas onCreateViewHolder(ViewGroup parent, int viewType) {
        View view=layoutInflater.inflate(R.layout.itencasas, parent, false);
        return new vhtodas(view);
    }

    @Override
    public void onBindViewHolder(final vhtodas holder, final int position) {

        YoYo.with(Techniques.Tada)
                .duration(700)
                .playOn(holder.itemView);

        final todascasas casaexata=casas.get(position);
        holder.txtLocalcasa.setText(casaexata.getLOCAL());
        holder.txtPrecocasa.setText(casaexata.getPRECO());
        String url=casaexata.getIMGURL();
        if(url!=null){
            imageLoader.get(url, new ImageLoader.ImageListener() {
                @Override
                public void onResponse(ImageLoader.ImageContainer response, boolean isImmediate) {
                    holder.imgcasa.setImageBitmap(response.getBitmap());
                }

                @Override
                public void onErrorResponse(VolleyError error) {
                    holder.imgcasa.setImageResource(R.drawable.logo);
                }
            });
        }
        holder.favoriteButton.setIcon(IconType.Heart);
        holder.favoriteButton.setIconSizeDp(25);
        holder.favoriteButton.setCircleEndColorRes(R.color.colorAccent);
        holder.favoriteButton.setExplodingDotColorsRes(R.color.colorPrimary,R.color.colorAccent);
        holder.favoriteButton.setLikeDrawableRes(R.drawable.heartliked);
        holder.favoriteButton.setUnlikeDrawableRes(R.drawable.heartunliked);

        holder.favoriteButton.setOnLikeListener(new OnLikeListener() {
            @Override
            public void liked(LikeButton likeButton) {
                firebaseutils.setlike(casaexata.getID());
                holder.favoriteButton.setLiked(true);
            }

            @Override
            public void unLiked(LikeButton likeButton) {
                firebaseutils.removelike(casaexata.getID());
                holder.favoriteButton.setLiked(false);
            }
        });
        firebaseutils.checklike(ctx,casaexata.getID(),holder.favoriteButton);
    }

    @Override
    public int getItemCount() {
        return casas.size();
    }

    static class vhtodas extends RecyclerView.ViewHolder implements View.OnClickListener{

        private ImageView imgcasa;
        private TextView txtLocalcasa;
        private TextView txtPrecocasa;
        private LikeButton favoriteButton;

        vhtodas(final View itemView) {
            super(itemView);

            imgcasa = (ImageView) itemView.findViewById(R.id.imgcasa);
            txtLocalcasa = (TextView) itemView.findViewById(R.id.txtlocalcasa);
            txtPrecocasa = (TextView) itemView.findViewById(R.id.txtprecocasa);
            favoriteButton = (LikeButton) itemView.findViewById(R.id.favbutton);


        }

        @Override
        public void onClick(View v) {

        }
    }
}

File: app/src/main/java/com/example/massa/luxvilla/adaptadores/adaptadorrvtodas.kt
package com.example.massa.luxvilla.adaptadores

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.android.volley.toolbox.ImageLoader
import com.daimajia.androidanimations.library.Techniques
import com.daimajia.androidanimations.library.YoYo
import com.example.massa.luxvilla.R
import com.example.massa.luxvilla.network.VolleySingleton
import com.example.massa.luxvilla.utils.casas
import com.example.massa.luxvilla.utils.firebaseutils
import com.like.IconType
import com.like.LikeButton
import com.like.OnLikeListener
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.itencasas.view.*
import java.util.ArrayList

/**
 * Created by massa on 27/10/2017.
 */
class adaptadorrvtodas(context: Context) : RecyclerView.Adapter<adaptadorrvtodas.vhtodas>() {

    private var casas = ArrayList<casas>()
    private val layoutInflater: LayoutInflater
    private val imageLoader: ImageLoader

    init {
        layoutInflater = LayoutInflater.from(context)
        val volleySingleton = VolleySingleton.getInstancia(context)
        imageLoader = volleySingleton.imageLoader
        ctx = context

    }

    fun setCasas(cs: ArrayList<casas>) {
        this.casas = cs
        notifyDataSetChanged()
    }

    fun removeAt(position: Int) {
        casas.removeAt(position)
        notifyItemRemoved(position)
        notifyItemRangeChanged(position, casas.size)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): vhtodas {
        val view = layoutInflater.inflate(R.layout.itencasas, parent, false)
        return vhtodas(view)
    }

    override fun onBindViewHolder(holder: vhtodas, position: Int) {

        YoYo.with(Techniques.Tada)
                .duration(700)
                .playOn(holder.itemView)

        val casaexata = casas[position]
        holder.txtLocalcasa.text = casaexata.local
        holder.txtPrecocasa.text = casaexata.preco
        val url = casaexata.imgurl
        if (url != null) {
            Picasso.with(ctx).load(url).error(R.drawable.logo).into(holder.imgcasa)
        }
        holder.favoriteButton.setIcon(IconType.Heart)
        holder.favoriteButton.setIconSizeDp(25)
        holder.favoriteButton.setCircleEndColorRes(R.color.colorAccent)
        holder.favoriteButton.setExplodingDotColorsRes(R.color.colorPrimary, R.color.colorAccent)
        holder.favoriteButton.setLikeDrawableRes(R.drawable.heartliked)
        holder.favoriteButton.setUnlikeDrawableRes(R.drawable.heartunliked)

        holder.favoriteButton.setOnLikeListener(object : OnLikeListener {
            override fun liked(likeButton: LikeButton) {
                firebaseutils.setlike(casaexata.id!!)
                holder.favoriteButton.setLiked(true)
            }

            override fun unLiked(likeButton: LikeButton) {
                firebaseutils.removelike(casaexata.id!!)
                holder.favoriteButton.setLiked(false)
            }
        })
        firebaseutils.checklike(ctx!!, casaexata.id!!, holder.favoriteButton)
    }

    override fun getItemCount(): Int {
        return casas.size
    }

    class vhtodas(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

         val imgcasa: ImageView
         val txtLocalcasa: TextView
         val txtPrecocasa: TextView
         val favoriteButton: LikeButton

        init {

            imgcasa = itemView.imgcasa
            txtLocalcasa = itemView.txtlocalcasa
            txtPrecocasa = itemView.txtprecocasa
            favoriteButton = itemView.favbutton


        }

        override fun onClick(v: View) {

        }
    }

    companion object {
        var ctx: Context? = null
    }
}
